iOSDFULibrary/Classes/Implementation/GenericDFU/DFUPeripheral.swift

```diff
@@ -290,6 +290,18 @@ internal class BaseDFUPeripheral<TD : BasePeripheralDelegate> : NSObject, BaseDF
         // Search for DFU service
         guard let dfuService = findDfuService(in: peripheral.services) else {
             logger.e("DFU Service not found")
+            
+            // Log what was found in case of an error
+            if let services = peripheral.services, services.isEmpty == false {
+                logger.d("The following services were discovered:")
+                services.forEach { service in
+                    logger.d(" - \(service.uuid.uuidString)")
+                }
+            } else {
+                logger.d("No services found")
+            }
+            logger.d("Is it the right device? If so, it may be caching issue. Try again after restarting Bluetooth. Make sure that your device has the Service Changed characteristic")
```

Comment:
Perhaps a better log that's a bit more formal and less uncertain ? something like that for example:
```
logger.d("This device might not be the correct target due to caching, toggle bluetooth for 10 seconds from settings, if t he issue persists, ensure the device contains the Service Changed characteristic")
```